presto-main/src/main/java/com/facebook/presto/sql/planner/Symbol.java

```diff
@@ -28,8 +29,16 @@
 
     public static Symbol from(Expression expression)
     {
-        checkArgument(expression instanceof SymbolReference, "Unexpected expression: %s", expression);
-        return new Symbol(((SymbolReference) expression).getName());
+        return optionalFrom(expression)
+                .orElseThrow(() -> new IllegalArgumentException("Unexpected expression: %s" + expression));
+    }
+
+    public static Optional<Symbol> optionalFrom(Expression expression)
```

Comment:
I don't think this is the right name for this method. Do we use such a naming convention already somewhere?
`fromSymbolReference`? `tryFrom`?